src/Plugins/ITKImageProcessing/src/ITKImageProcessing/Filters/Algorithms/ITKImportFijiMontage.cpp

```diff
@@ -0,0 +1,323 @@
+
+namespace
+{
+const Uuid k_ComplexCorePluginId = *Uuid::FromString("05cc618b-781f-4ac0-b9ac-43f26ce1854f");
+const Uuid k_ColorToGrayScaleFilterId = *Uuid::FromString("d938a2aa-fee2-4db9-aa2f-2c34a9736580");
+const FilterHandle k_ColorToGrayScaleFilterHandle(k_ColorToGrayScaleFilterId, k_ComplexCorePluginId);
+
+template <bool GenerateCache = true>
+class IOHandler
+{
+public:
+  IOHandler(ITKImportFijiMontage* filter, DataStructure& dataStructure, const ITKImportFijiMontageInputValues* inputs)
+  : m_Filter(filter)
+  , m_DataStructure(dataStructure)
+  , m_InputValues(inputs)
+  , m_Allocate(inputs->allocate)
+  , m_Cache(m_Filter->getCache())
+  {
+  }
+  ~IOHandler() = default;
+
+  IOHandler(const IOHandler&) = delete;
+  IOHandler(IOHandler&&) noexcept = delete;
+  IOHandler& operator=(const IOHandler&) = delete;
+  IOHandler& operator=(IOHandler&&) noexcept = delete;
+
+  [[nodiscard]] Result<> operator()()
+  {
+    m_InStream = std::ifstream(m_InputValues->inputFilePath, std::ios_base::binary);
+    if(!m_InStream.is_open())
+    {
+      return MakeErrorResult(-2013, fmt::format("Unable to open the provided file to read at path : {}", m_InputValues->inputFilePath.string()));
+    }
+
+    if(!m_Allocate)
+    {
+      FillCache();
+    }
+    else
+    {
+      return ReadImages();
+    }
+
+    return {};
+  }
+
+private:
+  // member variables
+  ITKImportFijiMontage* m_Filter;
+  DataStructure& m_DataStructure;
+  std::ifstream m_InStream;
+  const bool m_Allocate;
+  FijiCache& m_Cache;
+  const ITKImportFijiMontageInputValues* m_InputValues;
+
+  // -----------------------------------------------------------------------------
+  void ParseConfigFile()
+  {
+    bool dimFound = false;
+    bool dataFound = false;
+
+    std::string line;
+    while(std::getline(m_InStream, line))
+    {
+      line = StringUtilities::trimmed(line);
+
+      if(StringUtilities::starts_with(line, "dim =")) // found the dimensions
+      {
+        dimFound = true;
+        // Should check that the value = 2
+      }
+      if(StringUtilities::starts_with(line, "# Define the image coordinates"))
+      {
+        // Found the start of the data
+        dataFound = true;
+        break;
+      }
+      if(StringUtilities::starts_with(line, "#")) // comment line
+      {
+        continue;
+      }
+    }
+
+    if(!dimFound || !dataFound)
+    {
+      return;
+    }
+
+    // Example line to be parsed: slice_12.tif; ; (471.2965233276666, -0.522608066434236)
+    while(std::getline(m_InStream, line))
+    {
+      line = StringUtilities::trimmed(line);
+      if(line.empty())
+      {
+        continue;
+      }
+      std::vector<std::string> tokens = StringUtilities::split(line, ';');
+      if(tokens.size() != 3)
+      {
+        continue;
+      }
+      BoundsType bound;
+      std::stringstream path;
+      path << m_InputValues->inputFilePath.parent_path().string() << "/" << tokens[0];
+      bound.Filepath = fs::path(path.str());
```

Comment:
```suggestion
      bound.Filepath = m_InputValues->inputFilePath.parent_path() / tokens[0];
```
`std::filesystem::path::operator/` can concatenate paths.